rasterio/_base.pyx

```diff
@@ -1490,7 +1490,8 @@ def _transform(src_crs, dst_crs, xs, ys, zs):
     try:
         transform = OCTNewCoordinateTransformation(src, dst)
         transform = exc_wrap_pointer(transform)
-        exc_wrap_int(OCTTransform(transform, n, x, y, z))
+        # OCTTransform() returns TRUE/FALSE contrary to most GDAL API functions
+        exc_wrap_int(OCTTransform(transform, n, x, y, z) == 0)
```

Comment:
@rouault I see now. Because I had the return values flipped, errors weren't getting popped off the stack when they should and were getting picked up erroneously. And I can see how this depends on test order. The tests that we expect to pass have to be run after a test that we expect to raise an error.